clients/osv/src/funTest/kotlin/OsvServiceTest.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2022 EPAM Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import io.kotest.assertions.json.shouldEqualJson
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.result.beSuccess
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+
+import java.io.File
+
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+import org.ossreviewtoolkit.clients.osv.OsvService
+import org.ossreviewtoolkit.clients.osv.Package
+import org.ossreviewtoolkit.clients.osv.VulnerabilitiesForPackageRequest
+
+private val VULNERABILITY_FOR_PACKAGE_BY_COMMIT_REQUEST = VulnerabilitiesForPackageRequest(
+    commit = "6879efc2c1596d11a6a6ad296f80063b558d5e0f"
+)
+private val VULNERABILITY_FOR_PACKAGE_BY_NAME_AND_VERSION = VulnerabilitiesForPackageRequest(
+    pkg = Package(
+        name = "jinja2",
+        ecosystem = "PyPI"
+    ),
+    version = "2.4.1"
+)
+private val VULNERABILITY_FOR_PACKAGE_BY_INVALID_COMMIT_REQUEST = VulnerabilitiesForPackageRequest(
+    commit = "6879efc2c1596d11a6a6ad296f80063b558d5e0c"
+)
+
+class OsvServiceTest : StringSpec({
+    "getVulnerabilitiesForPackage()  returns the expected vulnerability when queried by commit" {
+        val expectedResult = getAssetAsString("vulnerabilities-by-commit-expected-result.json")
+
+        val result = OsvService().getVulnerabilitiesForPackage(VULNERABILITY_FOR_PACKAGE_BY_COMMIT_REQUEST)
+
+        result should beSuccess()
+        JSON.encodeToString(result.getOrNull()) shouldEqualJson expectedResult
```

Comment:
Now you could combine these to

        result.shouldBeSuccess {
            JSON.encodeToString(it) shouldEqualJson expectedResult
        }